src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -2766,15 +2766,51 @@ void Parser::parse_and_visit_switch(Parse_Visitor_Base &v) {
 
   bool keep_going = true;
   bool is_before_first_switch_case = true;
+  Token prev_token;
```

Comment:
Nit: I think this variable's name is bad. `prev_token` does not refer to the token just before the current token; it refers to a prior `case` or `default` token or the first token of a statement.

Maybe `previous_statement_first_token` is a better name?